src/types.ts

```diff
@@ -47,44 +47,43 @@ export interface InvestScript {
 // tokens and reward tokens should be sent to
 export interface DivestScript {
   readonly address: string;
-  readonly signature:
-    | 'function divest(address recipient, uint256[] calldata receiptAmounts, uint256[] calldata rewardAmounts) external payable'
-    | 'function divest(address recipient, uint256[] calldata receiptAmounts, uint256[] calldata rewardAmounts, bytes calldata data) external payable';
+  readonly signature: string;
 }
 
 export interface StrategyLeg {
-  action: 'Borrow' | 'Swap to' | 'Deposit for' | 'Deposit to' | 'Stake on';
-  token: string; // address of the token, to be looked up in the `tokens` mapping
-  platformId: number; // Platform ID of strategy target
-  logoURI?: string;
+  readonly action: 'Borrow' | 'Swap to' | 'Deposit for' | 'Deposit to' | 'Stake on';
+  readonly actionReceiver: string;
+  readonly logoURI?: string;
+  readonly platformId: number; // Platform ID of strategy target
+  readonly token: string; // address of the token, to be looked up in the `tokens` mapping
 }
 
-export interface InvestmentOpportunity {
-  readonly invest: InvestScript;
-  readonly divest: DivestScript;
-  readonly dataURI: string; // API endpoint to query for user data from, e.g. balances
-  readonly tokens: {
-    // arrays of token addresses, to be looked up in the `tokens` mapping
-    readonly input: string[]; // token addresses the user will deposit to enter this opportunity, and receipt upon exiting
-    readonly receipt: string[]; // receipt token addresses the user needs to exit this opportunity
-    readonly rewards: string[]; // rewards token addresses the user may claim when exiting this opportunity
-  };
-  readonly strategy?: StrategyLeg[];
+export interface MarketList {
+  readonly keywords?: string[];
+  readonly logoURI?: string;
+  readonly markets: MarketInfo[];
+  readonly name: string;
+  readonly tags?: Tags;
+  readonly timestamp: string;
+  readonly tokens: Record<string, TokenInfo>; // maps from checksummed token address to TokenInfo
+  readonly version: Version;
 }
 
-export interface ProtectionMarketInfo extends TokenInfo {
-  readonly underlying: string; // address of underlying, if `underlying === 0xEeee....EEeE`, underlying is ETH
-  readonly trigger: string; // trigger contract address for protection markets, or zero address for money markets
+export interface MarketInfo extends TokenInfo {
   readonly investmentOpportunity?: InvestmentOpportunity; // details on how protected investing using this market
+  readonly trigger: string; // trigger contract address for protection markets, or zero address for money markets
+  readonly underlying: string; // address of underlying, if `underlying === 0xEeee....EEeE`, underlying is ETH
```

Comment:
I know this is how it previously was, but now I think we should just spell out the whole address
```suggestion
  readonly underlying: string; // address of underlying, if `underlying === 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE`, underlying is ETH
```